yotierror/response.go

```diff
@@ -38,6 +38,8 @@ type Error struct {
 	message  string
 	Err      error
 	Response *http.Response
+	id       string
```

Comment:
just adding these new fields here won't work.
If you look here https://github.com/getyoti/yoti-go-sdk/blob/master/yotierror/response.go#L61C15-L137
You will see that the prev error format operated on the assumption that there is one HTTP status code that can be uniquely mapped to an error message.
That is not the case anymore as we how we can't rely on that mapping (and shouldn't do it in our code anymore). We should now just parse the body of response to get the error's details.
It will be simpler than before. 
We  probably need a new struct that describes the new error format, and prob it's best to have our own version of https://github.com/getyoti/yoti-go-sdk/tree/master/requests adequate for V2

Reply:
eg. bc the current code is still trying to do that mapping. instead of a properly formatted error you'll get an error with:

```
id = ""
status = 0
message = "401: unknown HTTP error - {"id":"8f6a9dfe72128de20909af0d476769b6","status":401,"error":"INVALID_REQUEST_SIGNATURE","message":"Invalid request signature"}"
```

Reply:
these fields have to be removed as we will have a new struct